moda-contracts/test/SplitsFactoryWithFork.t.sol

```diff
@@ -0,0 +1,198 @@
+        _splitMain = ISplitMain(_splitMainAddress);
+        _splitsFactory = new SplitsFactory(_splitMain, _modaRegistry);
+        vm.stopPrank();
+    }
+
+    function setupBeneficiaries() public {
+        _beneficiaries.push(_alice);
+        _beneficiaries.push(_bob);
+    }
+
+    // create
+
+    function create_split_setUp() public {
+        setupBeneficiaries();
+        vm.startPrank(_deployer);
+        _split = _splitsFactory.create(_beneficiaries);
+        vm.stopPrank();
+    }
+
+    function pay_split_setUp() public {
+        create_split_setUp();
+        vm.startPrank(_payer);
+        payable(_split).transfer(10 ether);
+        vm.stopPrank();
+    }
+
+    function test_distributeETH_from_SplitMain() public {
+        pay_split_setUp();
+
+        _beneficiaries.push(_modaTreasury);
+
+        _beneficiaries = _sortAddresses(_beneficiaries);
+
+        uint32[] memory percentAllocations = new uint32[](3);
+
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            if (_beneficiaries[i] == _modaTreasury) {
+                percentAllocations[i] = 100_000;
+            } else {
+                percentAllocations[i] = 450_000;
+            }
+        }
+
+        uint256 gasBefore = gasleft();
+
+        vm.startPrank(_distributor);
+        _splitMain.distributeETH(_split, _beneficiaries, percentAllocations, 10_000, _distributor);
+        vm.stopPrank();
+
+        uint256 gasAfter = gasleft();
+
+        console2.log("Gas used: %s", gasBefore - gasAfter);
+
+        uint256 balanceAlice = _splitMain.getETHBalance(_alice);
+        uint256 balanceBob = _splitMain.getETHBalance(_bob);
+        uint256 balanceTreasury = _splitMain.getETHBalance(_modaTreasury);
+        uint256 distributorBalance = _splitMain.getETHBalance(_distributor);
+
+        // 1% to the distributor => 10 ether * 0.01 = 0.1 ether
+        assertEq(distributorBalance, 0.1 ether);
+        // 45% each to Alice and Bob => 9.9 ether * 0.45 = 4.455 ether
+        assertEq(balanceAlice, 4.455 ether);
+        assertEq(balanceBob, 4.455 ether);
+        // 10% to the treasury => 9.9 ether * 0.1 = 0.99 ether
+        assertEq(balanceTreasury, 0.99 ether);
+    }
+
+    function withdraw_setUp() public {
+        pay_split_setUp();
+        _beneficiaries.push(_modaTreasury);
+
+        _beneficiaries = _sortAddresses(_beneficiaries);
+
+        uint32[] memory percentAllocations = new uint32[](3);
+
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            if (_beneficiaries[i] == _modaTreasury) {
+                percentAllocations[i] = 100_000;
+            } else {
+                percentAllocations[i] = 450_000;
+            }
+        }
+
+        vm.startPrank(_distributor);
+        _splitMain.distributeETH(_split, _beneficiaries, percentAllocations, 10_000, _distributor);
+        vm.stopPrank();
+    }
+
+    function test_withdraw_from_splitMain() public {
+        vm.stopPrank();
```

Comment:
This seems strange that a prank is in-progress. This shouldn't be the case. No need to `stopPrank` in the beginning of a test. It should be a clean start